File: apis/telegram.py
import os
import json
import urllib
import logging

from google.appengine.api import urlfetch
from google.appengine.api.urlfetch_errors import DeadlineExceededError

BASE_URL = 'https://api.telegram.org/bot{token}/{method}'
TOKEN = os.environ['TELEGRAM_BOT_TOKEN']


def call_method(method, data, deadline=10):
  data = json.dumps(data)
  try:
    result = urlfetch.fetch(
        BASE_URL.format(token=TOKEN, method=method),
        payload=data,
        method=urlfetch.POST,
        deadline=deadline,
        headers={'Content-Type': 'application/json'})
  except DeadlineExceededError as e:
    logging.exception(e)
    return None
  if result.status_code == 200:
    return json.loads(result.content)
  else:
    logging.error(result.content)
    return None


def send_message(chat_id, text, reply_markup=None, parse_mode='HTML',
                 disable_notification=True):
  message = {
    'chat_id': chat_id,
    'text': text,
    'parse_mode': parse_mode,
    'disable_notification': disable_notification
  }

  if reply_markup:
    message['reply_markup'] = reply_markup

  return call_method('sendMessage', message)

def send_photo(chat_id, photo, text='', reply_markup=None, parse_mode='HTML',
               disable_notification=True):
  message = {
    'chat_id': chat_id,
    'photo': photo,
    'caption': text,
    'parse_mode': parse_mode,
    'disable_notification': disable_notification
  }

  if reply_markup:
    message['reply_markup'] = reply_markup

  return call_method('sendPhoto', message, deadline=30)

def send_media_group(chat_id, photos, text='', reply_markup=None,
                     parse_mode='HTML', disable_notification=True):
  media = list({'type': 'photo', 'media': photo, 'caption': text, 'parse_mode': parse_mode} for photo in photos)
  message = {
    'chat_id': chat_id,
    'media': media,
    'disable_notification': disable_notification
  }

  if reply_markup:
    message['reply_markup'] = reply_markup

  return call_method('sendMediaGroup', message, deadline=30)

File: src/apis/telegram.ts
import axios from 'axios';

const BASE_URL = 'https://api.telegram.org/bot'
const TOKEN = process.env.TELEGRAM_BOT_TOKEN;



async function call(method: string, data: any, timeout = 10000): Promise<any> {
  try {
    const result = await axios.post(`${BASE_URL}${TOKEN}/${method}`, data, {
      timeout, headers: {
        'Content-Type': 'application/json'
      }
    });
    if (result.status === 200) return result.data
  } catch (error) {
    console.error(error)
  }
  return null;
}

export async function sendMessage(chatId: string, text: string, replyMarkup:any = undefined, parseMode = 'HTML',
  disableNotification = true): Promise<any> {
  const message = {
    'chat_id': chatId,
    'text': text,
    'parse_mode': parseMode,
    'disable_notification': disableNotification,
    reply_markup: replyMarkup
  }

  return call('sendMessage', message)
}


export async function sendPhoto(chatId: string, photo: any, text = '', replyMarkup:any = undefined, parseMode = 'HTML',
  disableNotification = true): Promise<any> {
  const message = {
    'chat_id': chatId,
    photo,
    'caption': text,
    'parse_mode': parseMode,
    'disable_notification': disableNotification,
    reply_markup: replyMarkup
  }

  return call('sendPhoto', message, 30000)
}

export async function sendVideo(chatId: string, video: any, text = '', replyMarkup:any = undefined, parseMode = 'HTML',
  disableNotification = true): Promise<any> {
  const message = {
    'chat_id': chatId,
    video,
    'caption': text,
    'parse_mode': parseMode,
    'disable_notification': disableNotification,
    reply_markup: replyMarkup
  }

  return call('sendVideo', message, 30000)
}

export async function sendMediaGroup(chatId: string, photos: any[], text = '', replyMarkup:any = undefined, parseMode = 'HTML',
  disableNotification = true): Promise<any> {
  const media = photos.map(photo => ({ 'type': 'photo', 'media': photo, 'caption': text, 'parse_mode': parseMode }))
  const message = {
    'chat_id': chatId,
    media,
    'disable_notification': disableNotification,
    reply_markup: replyMarkup
  }

  return call('sendMediaGroup', message, 30000)
}